### DISCUSSION TITLE
newbie question: Can sunshine stream from  KVM *host* while using GPU passthrough?

### DISCUSSION THREAD

**Discussion by huntercr42** [LizardByte/community#377] [Sunshine]
Newbie sunshine user here. I am not new to *nix, but definitely new to modern linux devices and graphics stack, so I don't know if this question is inherently silly or what...

Here's the scenario: 

Headless Linux Host with KVM  and a single GPU
 1. Linux Guest VM using GPU passthrough 
 2. Windows Guest VM using GPU passthrough 
 
I have seen several threads about using sunshine inside of those VMs, but I was curious about simplifying/lowering the overhead if possible. I am only going to be using 1 VM at a time, so can sunshine execute at the host level and still be able to see what is passing through to the GPU on the running VM? I'm not sure if that would keep any of the low level PCI device(s) still "In use" and cause the passthrough to fail? Is this even architecturally possible? 

Now that. think about this, it would also have to support "passthrough" of controller/keyboard/mouse input ( not to mention audio output ) . I am starting to think this is unlikely, but keeping fingers crossed! Thank you!

**Comment by ivdimitro** [comment]
Short answer - NO. The whole point of a VM is to be isolated from the host. That's why you need SPICE or sunshine to get the VM screen.
You can achieve what you want with [https://looking-glass.io](url)

**Reply by huntercr42** [reply]
Thank you for the response... I kept on learning after I wrote this post and learned  a lot more recently about the linux graphics stack I can see how tis is not normally possible. 
For the record, I actually have got it working though. Mine is a special case, I think because I am using an Intel chipset with GVT-g, which allows me to keep the Linux host ( Ubuntu ) up and running VMs in full screen so sunshine can capture it. I don't think this approach would have worked if I used traditional PCI passthrough.

I am going to take look ( no pun intended ) and looking glass more closely to see if it will meet my needs.
I may be wrong, but as I understand it, looking glass passes video between VMto VM and Host to VM only.  I don't think it will do Remote client  to VM streaming.

[a comment by an automated account is not shown]
